bigquery/main.go

```diff
@@ -56,17 +75,47 @@ func (bq *bqNotifier) SetUp(ctx context.Context, cfg *notifiers.Config, _ notifi
 	if err != nil {
 		return fmt.Errorf("failed to make a CEL predicate: %v", err)
 	}
-	_, ok := cfg.Spec.Notification.Delivery["table"].(string)
+	parsed, ok := cfg.Spec.Notification.Delivery["table"].(string)
 	if !ok {
 		return fmt.Errorf("Expected table string: %v", cfg.Spec.Notification.Delivery)
 	}
 
+	// Initialize client
 	bq.filter = prd
 	bq.client, err = bigquery.NewClient(ctx, projectID)
 	if err != nil {
 		return fmt.Errorf("Failed to initialize bigquery client: %v", err)
 	}
 
+	// Extract dataset id and table id from config
+	rgp, _ := regexp.Compile(".*/.*/.*/(.*)/.*/(.*)")
+	rs := rgp.FindStringSubmatch(parsed)
+	bq.dataset = bq.client.Dataset(rs[1])
+	bq.table = bq.dataset.Table(rs[2])
+
+	// Check for existence of dataset, create if false
+	_, err = bq.dataset.Metadata(ctx)
+	if err != nil {
+		log.Infof("Error obtaining dataset metadata: %v", bq.dataset)
+		if err := bq.dataset.Create(ctx, &bigquery.DatasetMetadata{}); err != nil {
```

Comment:
Fix: Can you fill in the Name and Description fields please in the DatasetMetadata?

https://pkg.go.dev/cloud.google.com/go/bigquery?tab=doc#DatasetMetadata

Reply:
Resolved!